apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/settings/update-key-remaining.tsx

```diff
@@ -219,34 +167,40 @@ export const UpdateKeyRemaining: React.FC<Props> = ({ apiKey }) => {
                         className="w-full"
                         type="number"
                         {...field}
-                        value={form.watch("refill.interval") === "none" ? undefined : field.value}
+                        value={form.getValues("limitEnabled") ? field.value : undefined}
                       />
                     </FormControl>
                     <FormDescription>
                       Enter the number of uses to refill per interval.
                     </FormDescription>
-                    <FormMessage defaultValue="Please enter a value if interval is selected" />
+                    <FormMessage />
                   </FormItem>
                 )}
               />
               <FormField
                 control={form.control}
-                disabled={form.watch("refill.interval") !== "monthly"}
+                disabled={
+                  form.watch("remaining") === undefined ||
+                  form.watch("remaining") === null ||
```

Comment:
If remaining being `0` is not a concern let's just do `!form.watch("remaining")`